dialogue-blocking-channels/src/main/java/com/palantir/dialogue/blocking/CallingThreadExecutor.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.blocking;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.dialogue.Request;
+
+public interface CallingThreadExecutor {
+
+    ListenableFuture<?> submit(Runnable task);
```

Comment:
I think we only use the result to cancel the submitted task, we can probably get away with `Future<?>` or even a simpler disposer/runnable since the value is only used to provide input (cancel with interruption) and not output (`future.get()`).

```suggestion
    Future<?> submit(Runnable task);
```

Reply:
I had it as future before, but writing tests required some gymnastics, since I'm not that familiar with the CompletableFuture API as much. Since we use ListenableFuture everywhere else, I'd rather we keep it as is.